aidial_sdk/utils/_cancel_scope.py

```diff
@@ -0,0 +1,72 @@
+import asyncio
+from asyncio import exceptions
+from typing import Optional, Set
+
+
+class CancelScope:
+    """
+    Async context manager that enforces cancellation of all tasks created within its scope when either:
+    1. the parent task has been cancelled or has thrown an exception or
+    2. any of the tasks created within the scope has thrown an exception.
+    """
+
+    def __init__(self):
+        self._tasks: Set[asyncio.Task] = set()
+        self._on_completed_fut: Optional[asyncio.Future] = None
+        self._cancelling: bool = False
+
+    async def __aenter__(self):
+        return self
+
+    async def __aexit__(self, et, exc, tb):
```

Comment:
Let's use the same naming, that [official PEP](https://peps.python.org/pep-0492/#asynchronous-context-managers-and-async-with) introduces:

```python
    async def __aexit__(self, exc_type, exc, tb):
        ...
```

It's quite easy to confuse `et` with something else, then exception type.

Reply:
Fixed